src/traces/bar/cross_trace_calc.js

```diff
@@ -108,7 +108,7 @@ function setGroupPositions(gd, pa, sa, calcTraces) {
     collectExtents(calcTraces, pa);
 }
 
-function initBase(gd, pa, sa, calcTraces) {
```

Comment:
No need to do this.

Please keep the more-general `(gd, pa, sa, calcTraces)` call signatures for all methods in this file. that way they can potentially be easily reused by other trace module.